x-pack/plugins/data_enhanced/server/search/session/get_search_status.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ApiResponse } from '@elastic/elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { isAsyncSearchStatusResponse, SearchStatus } from './types';
+
+export interface SearchStatusInfo {
+  status: SearchStatus;
+  error?: string;
+}
+
+export async function getSearchStatus(
+  client: ElasticsearchClient,
+  asyncId: string
+): Promise<SearchStatusInfo> {
+  try {
+    // TODO: Handle strategies other than KQL
+    const response: ApiResponse<any> = await client.asyncSearch.status({ id: asyncId });
+    if (isAsyncSearchStatusResponse(response)) {
+      if ((response.is_partial && !response.is_running) || response.completion_status > 200) {
+        return {
+          status: SearchStatus.ERROR,
+          error: `Search completed with a ${response.completion_status} status`,
```

Comment:
Will this error be surfaced in the UI at some point? If so, we need to i18n it.

Reply:
Management UI will show it @tsullivan 
Will `i18n`